cmd/init-osm-controller/init-osm-controller.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"time"
+
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha1"
+	"github.com/openservicemesh/osm/pkg/cli"
+	configClientset "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned"
+	"github.com/openservicemesh/osm/pkg/logger"
+	"github.com/openservicemesh/osm/pkg/version"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const meshConfigName = "osm-mesh-config"
+
+var settings = cli.New()
+
+var (
+	flags = pflag.NewFlagSet("init-osm-controller", pflag.ExitOnError)
+	log   = logger.New("osm-controller/main")
+
+	meshName           string // An ID that uniquely identifies an OSM instance
+	kubeConfigFilePath string
+	osmNamespace       string
+
+	fetchMeshConfigMaxRetry      = 5
+	fetchMeshConfigRetryInterval = 5
+)
+
+func init() {
+	if err := logger.SetLogLevel("info"); err != nil {
+		log.Fatal().Err(err).Msg("Error setting log level")
+	}
+
+	flags.StringVar(&meshName, "mesh-name", "", "OSM mesh name")
+	flags.StringVar(&kubeConfigFilePath, "kubeconfig", "", "Path to Kubernetes config file.")
+	flags.StringVar(&osmNamespace, "osm-namespace", "", "Namespace to which OSM belongs to.")
+}
+
+func parseFlags() error {
+	if err := flags.Parse(os.Args); err != nil {
+		return err
+	}
+	_ = flag.CommandLine.Parse([]string{})
+	return nil
+}
+
+func main() {
+	log.Info().Msgf("Starting init-osm-controller %s; %s; %s", version.Version, version.GitCommit, version.BuildDate)
+
+	if err := parseFlags(); err != nil {
+		log.Fatal().Err(err).Msg("Failed parsing cmd line arguments")
+	}
+
+	// Initialize kube config and client
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigFilePath)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Failed creating kube config (kubeconfig=%s)", kubeConfigFilePath)
+	}
+
+	configClient, err := configClientset.NewForConfig(kubeConfig)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Could not access Kubernetes cluster, check kubeconfig.")
+		return
+	}
+
+	var retry int
+	for retry = 0; retry < fetchMeshConfigMaxRetry; retry++ {
+		if _, err := configClient.ConfigV1alpha1().MeshConfigs(settings.Namespace()).Get(context.TODO(), meshConfigName, v1.GetOptions{}); err != nil {
+			// MeshConfig does not exist. Create a default one.
+			log.Info().Msgf("MeshConfig not exist. Trying create the default MeshConfig named %s into namespace %s", meshConfigName, osmNamespace)
+
+			if err := createDefaultMeshConfig(configClient); err != nil {
```

Comment:
There could be a race here with multiple controller pods starting up attempting to create the default MeshConfig CRD.

Could you follow a similar pattern here as done for loading the OSM CA bundle from a k8s secret here:
https://github.com/openservicemesh/osm/blob/77531d59eb10fc2cbf9a74fab3a7a5389445fce7/pkg/certificate/providers/config.go#L181-L188